test/v3/ERC721CollectionFactoryV3.spec.js

```diff
@@ -381,6 +384,102 @@ describe('Factory V3', function () {
       expect(isCollectionFromFactory).to.be.eq.BN(true)
     })
 
+    it('should create 3 collections and upgrade all of them', async function () {
+      expect(await factoryContract.collectionsSize()).to.be.eq.BN(0)
+
+      const createCollection = async (id) => {
+        const salt = web3.utils.randomHex(32)
+
+        const data = getInitData({
+          name: name + id,
+          symbol: symbol + id,
+          baseURI: baseURI + id,
+          creator: user,
+          shouldComplete: true,
+          isApproved: true,
+          items: ITEMS,
+          rarities: raritiesContract.address,
+        })
+
+        await factoryContract.createCollection(salt, data, fromFactoryOwner)
+
+        return factoryContract.getAddress(salt, factoryOwner, data)
+      }
+
+      const collectionAddress1 = await createCollection('-1')
+      const collectionAddress2 = await createCollection('-2')
+      const collectionAddress3 = await createCollection('-3')
+
+      expect(await factoryContract.collectionsSize()).to.be.eq.BN(3)
+
+      const collectionUpgradeImplementation =
+        await DummyERC721CollectionV2Upgrade.new()
+
+      await upgradeableBeaconContract.upgradeTo(
+        collectionUpgradeImplementation.address
```

Comment:
Can we check if the new function and the new variable does not exist before the upgrade and if the collections name, symbol, and URI  are the ones expected too?

Reply:
Done